### PR TITLE
Feat/events linkasset

### LINKED ISSUE #122: [Events] before/after LinkAsset
# Description

Add events before/after a Device is linked to asset

Equals to device-manager/device:beforeLinkAsset

- [x] device-manager:device:link-asset:before => { device, asset }

Equals to device-manager/device:afterLinkAsset

- [x] device-manager:device:link-asset:after => { device, asset }

### PR DESCRIPTION
**Pull request by rolljee**
# Description

This PR allow the user to connect to two different events that are triggered before and after an asset is linked to a device.

Two events when this action is called, allowing to modify the device before it is linked to the asset:

## Events

Two events when this action is called, allowing to modify the device before it is linked to the asset:

```js
app.pipe.register('device-manager:device:link-asset:before', async ({ device, asset }) => {
  app.log.debug('before link-asset trigered');
  return { device, asset };
})
app.pipe.register('device-manager:device:link-asset:after', async ({ device, asset }) => {
  app.log.debug('after link-asset trigered');
  return { device, asset };
})
```

close #122

### CODE REVIEW

**Comment by rolljee** [comment]
missing new tests for events

File: lib/core-classes/DeviceService.ts [hunk 1]
```diff
@@ -286,6 +295,11 @@ export class DeviceService {
       results.errors.concat(updatedDevice.errors, updatedDevice.errors);
     }
 
+    await global.app.trigger(
```
Comment (Aschen) [inline lib/core-classes/DeviceService.ts:298]: Same comment here

**Review by Aschen — CHANGES_REQUESTED** [review]
I think we should emit 1 event per asset/device pair.

Also you didn't add functional or unit tests to this PR

**Comment by rolljee** [comment]
functional tests are coming in the next commit :+1:

**Review by Aschen — APPROVED** [review]

File: lib/core-classes/DeviceService.ts [hunk 2]
```diff
@@ -495,8 +516,8 @@ export class DeviceService {
 
     if (deviceDocuments.length <= limit) {
       const { successes, errors } = await writer(deviceDocuments);
-      results.successes.push(successes);
-      results.errors.push(errors);
+      results.successes.push(...successes);
```
[the remark this answers is not in the document]
Comment (rolljee) [reply lib/core-classes/DeviceService.ts:519]: I have a PR that modify this behavior, I am gonna merge this one, and get things done in the orther PR

**Pull request merged**